libs/sdk-core/src/breez_services.rs

```diff
@@ -1924,36 +1924,58 @@ impl BreezServices {
 
     /// Registers for lightning payment notifications. When a payment is intercepted by the LSP
     /// to this node, a callback will be triggered to the `webhook_url`.
+    ///
+    /// Note: these notifications are registered for all LSPs (active and historical) with whom
+    /// we have a channel.
     async fn register_payment_notifications(&self, webhook_url: String) -> SdkResult<()> {
         let message = webhook_url.clone();
         let sign_request = SignMessageRequest { message };
         let sign_response = self.sign_message(sign_request).await?;
-        let lsp_info = self.lsp_info().await?;
-        self.lsp_api
-            .register_payment_notifications(
-                lsp_info.id,
-                lsp_info.lsp_pubkey,
-                webhook_url.clone(),
-                sign_response.signature,
-            )
-            .await?;
+        for lsp_info in get_notification_lsps(
+            self.persister.clone(),
+            self.lsp_api.clone(),
+            self.node_api.clone(),
+        )
+        .await?
+        {
+            self.lsp_api
+                .register_payment_notifications(
+                    lsp_info.id,
+                    lsp_info.lsp_pubkey,
+                    webhook_url.clone(),
+                    sign_response.signature.clone(),
+                )
+                .await?;
```

Comment:
There's some magic we have to do for error handling here. I think it doesn't matter when we fail to register notifications for an old LSP, but it does matter whn we try to register for notifications on a new LSP. Perhaps the function should return a response with the LSPs it registered and the ones where registration failed.

Reply:
Based on the original issue, IMO both cases are just as important: registering notifs for the active or a historical LSP.

However I did update the loop, such that it doesn't return immediately if a register call fails: dfca1fee5a3b51db6a9c3fb0326886e50be19c51